### PR TITLE
Creates Executors tab for Jobs UI

### PR DESCRIPTION
**Pull request by karenfeng**
This create an Executors Tab for the Jobs UI and should solve [SPARK-808](https://spark-project.atlassian.net/browse/SPARK-808).
- Moved summary from Storage to Executors Tab
- Created executor tables for RDD blocks, disk and memory usage
  executors tab

### CODE REVIEW

**Comment by mateiz** [comment]
Jenkins, add to whitelist

**Comment by mateiz** [comment]
Hey Karen, this looks good, but instead of a separate table for each executor, can you make it have just one big table with ID, address, RDD blocks, memory and disk as columns? That would allow us to sort the executors by these things. Look at how it's done for the tasks on the Stage page, for example.

**Comment by karenfeng** [comment]
Thanks @mateiz. Does this work? (On second thought, maybe I should separate the memory used from the maximum memory)
executors tab

**Comment by mateiz** [comment]
I think it's fine to show it this way, but show it as memoryUsed / memoryTotal instead of the current version, which is remaining / used.

File: core/src/main/scala/spark/ui/exec/ExecutorsUI.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      </div>;
+
+    headerSparkPage(content, sc, "Executors", Executors)
+  }
+
+  def getExecInfo(a: Int): Seq[String] = {
+    val execId = sc.getExecutorStorageStatus(a).blockManagerId.executorId
+    val hostPort = sc.getExecutorStorageStatus(a).blockManagerId.hostPort
+    val memUsed = Utils.memoryBytesToString(sc.getExecutorStorageStatus(a).maxMem)
+    val maxMem = Utils.memoryBytesToString(sc.getExecutorStorageStatus(a).memUsed())
```
Comment (mateiz) [inline core/src/main/scala/spark/ui/exec/ExecutorsUI.scala]: Maybe the problem is that you are using "used" for "max" here and the other way around above.

**Comment by karenfeng** [comment]
@mateiz, the Executors page now displays the number of complete/total (failed) tasks.
executors page

**Comment by mateiz** [comment]
Hey, tanks for the changes. A couple of other comments:
- For the tasks, split it into three columns: Total Tasks, Tasks Complete, and Tasks Failed. This is because we might want to sort by these different categories.
- For the memory column, add a `sorttable_customkey` attribute to the `<td>` element so that we will sort the column by the numeric value of the memory rather than the string that will be printed. For example, "2 MB" should show up before "1 GB", but if you compare strings instead of numbers, the "1 GB" would show up first. Look at `spark.deploy.worker.ui.IndexPage` on line 98 to see how we add this attribute.

Other than that, it's looking really good.

**Comment by karenfeng** [comment]
executors tab

**Comment by mateiz** [comment]
Thanks Karen. Merging this in.

**Pull request merged**